### PR TITLE
chore: Update documentation for the governed-documents

### LINKED ISSUE #1146: Update governed-docs documentation
## Issue description / feature objectives
Currently, the documentation leaves a lot of questions on the versioning workflow for the governed documents. The task is to update the documentation for the versioning of the governed-docs.

## Completion Criteria
- Documentation provides a clear and illustrative steps for the user to follow when creating a new template version.

### PR DESCRIPTION
**Pull request by enikonovad**
Signed-off-by: Ekaterina Nikonova <enikonovad@gmail.com>

## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary
Updated documentation on the versioning of the governed documents

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

Closes #1146

### CODE REVIEW

File: docs/trestle_author.md [hunk 1]
```diff
@@ -127,6 +131,56 @@ The data presents as a set of key-value pairs which can be edited (see below). T
...
+### Prerequisite:
+
+This section assumes that you have an existing Trestle workspace initialized, with existing template and governed documents. Please follow steps in the section `# Getting Started` if you don't have one.
+
+Trestle provides the capability to version the templates and the documents via `x-trestle-template-version` field in the header.
+
+Consider an example where we have a governed document called `Decision 1` that we now need to update to contain a new header field `approved-status` and a new required heading `Heading n` at the end, as demonstrated in the Figure below:
+![Decision document update](assets/template_versioning.png)
+
+The intented workflow in this scenario is to:
```
Comment (fsuits) [inline docs/trestle_author.md:145]: intended

File: docs/tutorials/work_with_authoring_versions.md [hunk 2]
```diff
@@ -5,38 +5,51 @@ authoring content using the [compliance-trestle](https://ibm.github.io/complianc
 
 ## Step 1: Create a trestle workspace if you don't have one already
 
+This will initialize Trestle workspace that would contain all your templates and governed documents.
+
```
Comment (fsuits) [inline docs/tutorials/work_with_authoring_versions.md:9]: This intializes a Trestle workspace that will contain all ....

File: docs/tutorials/work_with_authoring_versions.md [hunk 3]
```diff
@@ -5,38 +5,51 @@ authoring content using the [compliance-trestle](https://ibm.github.io/complianc
...
 ```bash
 mkdir my_workspace
 cd my_workspace
 trestle init
 >>> Initialized trestle project successfully in [user_path]/my_workspace
 ```
 
+The templates will be located in the `/my_workspace/.trestle/author/`. Please note that if you use MacOS or Linux distribution, by default `.trestle` will be hidden. You will need to reconfigure your system to show hidden folders.
```
Comment (fsuits) [inline docs/tutorials/work_with_authoring_versions.md:17]: You will need to reconfigure your system to show hidden folders if you want to see them listed.

File: docs/tutorials/work_with_authoring_versions.md [hunk 4]
```diff
@@ -5,38 +5,51 @@ authoring content using the [compliance-trestle](https://ibm.github.io/complianc
...
+>>> Task directory is decisions
 ```
 
 Note: version 0.0.1 is reserved for unversioned documents.
 
 ## Step 3: Create a new document using a particular template version
 
+Now that we have a template, we would like to create an instance of this template that would actually have content in it.
+This steps will create a complete copy of the template of version `0.0.2` that you can fill in with custom content.
+
```
Comment (fsuits) [inline docs/tutorials/work_with_authoring_versions.md:38]: These steps will...

**Review by fsuits — APPROVED** [review]
LGTM after correcting a few typos and language changes.

File: docs/tutorials/work_with_authoring_versions.md (hunk 4 above)
Comment (enikonovad) [reply docs/tutorials/work_with_authoring_versions.md]: Done

**Pull request merged**